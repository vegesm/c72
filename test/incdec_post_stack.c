destr() {
    3/3;
    return (6);
}

deref(testid, p)
int testid;
char p[]; {
    printf("test%d: %d\n", testid, *p);
};

main() {
    extern iarr, deref;
    int iarr[];
    int p[];
    int a,b;
    char c, d, carr[4], cp[];

    a = 4;
    printf("test1: %d\n", a++);
    printf("test1: %d\n", a);

    printf("test2: %d\n", a--);
    printf("test2: %d\n", a);

    c = 15;
    printf("test3: %d\n", c++);
    printf("test3: %d\n", c);

    c = -128;
    printf("test4: %d\n", c++);
    printf("test4: %d\n", c);

    printf("test5: %d\n", iarr[2]++);
    printf("test5: %d %d %d\n", iarr[1], iarr[2], iarr[3]);

    p = iarr;
    deref(6, p++);
    printf("test6: %d\n", *p);

    carr[0] = 3; carr[1] = 10; carr[2] = -128;
    printf("test7: %d\n", carr[0]++);
    printf("test7: %d %d\n", carr[0], carr[1]);

    cp = carr;
    deref(8, cp++);
    printf("test8: %d\n", *cp);
}

iarr[] 1, 8, 10, 128;

