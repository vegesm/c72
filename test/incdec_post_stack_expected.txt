test1: 4
test1: 5
test2: 5
test2: 4
test3: 15
test3: 16
test4: -128
test4: -127
test5: 10
test5: 8 11 128
test6: 1
test6: 8
test7: 3
test7: 4 10
test8: 4
test8: 10
